### DISCUSSION TITLE
Way to cast value

### DISCUSSION THREAD

**Discussion by ValYouW** [react-querybuilder/react-querybuilder#744] [Q&A]
Hi,

Is there a way to cast the value from a "select" editor?
Say I want to show a true/false single select component, currently the returned value is a string of "true"/"false", what is the best way to cast it to Boolean?
Would have been nice if a Field had a "cast" property, something like the "validator" property where we could supply a function to cast the type.
Thx.

--- UPDATE ---
I did something like this and I wonder whether it is legit.

My fields definition:
Added a `datatype` property.

```js
const fields: Field[] = [
  { 
     name: 'gender',
     label: 'Gender',
     valueEditorType: 'select',
     datatype: 'number',
     values: [
      {label: 'Male', name: 0},
      {label: 'Female', name: 1},
      {label: 'Other', name: 2},
     ]
  },
  { 
    name: 'member',
    label: 'Member',
    valueEditorType: 'select',
    datatype: 'boolean',
    values: [
     {label: 'true', name: true},
     {label: 'false', name: false},
    ]
 },
];
```

Then in a custom value editor I use the `datatype`:
```js
let cast = {
	number: v => Number(v),
	boolean: v => {
		return v === 'true' ? true : false;
	}
};

function CustomValueEditor(props) {
	let {  datatype } = props.fieldData;
	let { handleOnChange, ...rest } = props;

	let onChange = useCallback(val => {
		if (datatype && typeof cast[datatype] === 'function') {
			if (Array.isArray(val)) {
				val = val.map(v => cast[datatype](v));
			} else {
				val = cast[datatype](val);
			}
		}

		handleOnChange(val);
	}, [datatype, handleOnChange]);

	return <ValueEditor {...rest} handleOnChange={onChange} />;
};
```

**Comment by jakeboone02** [comment]
I'd probably end up writing it a little differently, but if that works for you then yeah, it looks "legit." There are some other examples of using a `datatype` property [here](https://react-querybuilder.js.org/docs/tips/custom-with-fallback) and [here](https://react-querybuilder.js.org/docs/tips/managing-operators#getoperators-prop).

**Reply by ValYouW** [reply]
Thx
